mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/voice/VoiceInputCombinator.kt

```diff
@@ -20,36 +26,76 @@ object VoiceInputCombinator {
    */
   // TODO : Internationalization
   private val rank =
+  // Try to transformed into a our chess word
+  combine(
+              convertToken(ChessSpeechEnglishDictionary.chessPieces),
+              token().map { token ->
+                rulesSet
+                    .map { chessRule ->
+                      // Apply rule
+                      val res = chessRule.rule(token)
+                      res
+                    }
+                    .firstOrNull { res ->
+                      // Get the 1st value that have been transformed
+                      res != null
+                    }
+              })
+          .map {
+            when (it) {
+              "king" -> Rank.King
+              "queen" -> Rank.Queen
+              "rook" -> Rank.Rook
+              "bishop" -> Rank.Bishop
+              "knight" -> Rank.Knight
+              "pawn" -> Rank.Pawn
+              else -> null
+            }
+          }
+
+  val column =
+      convertToken(ChessSpeechEnglishDictionary.letters).filter { it in 'a'..'h' }.map {
+        it!! - 'a'
+      }
```

Comment:
Same as below comment

Reply:
Fix in https://github.com/epfl-SDP/android/pull/311/commits/b89ed32112920433b6c2828bcbcff924909337e1